vantage6/node/proxy_server.py

```diff
@@ -1,262 +1,339 @@
-    unencrypted = request.get_json()
-    organizations = unencrypted.get("organizations", None)
+def make_proxied_request(endpoint: str) -> Response:
+    """
+    Helper to create proxies requests to the central server.
 
-    if not organizations:
-        log.error("No organizations found?!")
-        return
-
-    n_organizations = len(organizations)
-    log.debug(f"{n_organizations} organizations, attemping to encrypt")
-    encrypted_organizations = []
-
-    for organization in organizations:
-        input_ = organization.get("input", {})
-        if not input_:
-            log.error("No input for organization?!")
-            return
-
-        organization_id = organization.get("id", None)
-        log.debug(f"retreiving public key of org={organization_id}")
-
-        # retrieve public key of the organization
-        response = requests.get(
-            f"{url}/organization/{organization_id}",
-            headers={'Authorization': auth}
-        )
+    Parameters
+    ----------
+    endpoint: str
+        endpoint to be reached at the vantage6-server
 
-        public_key = response.json().get("public_key")
+    Returns
+    -------
+    flask.Response
+        Response from the vantage6-server
+    """
+    present = 'Authorization' in request.headers
+    headers = {'Authorization': request.headers['Authorization']} if present \
+        else None
 
-        # Simple JSON (only for unencrypted collaborations)
-        # if isinstance(input_, dict):
-        #     input_ = bytes_to_base64s(
-        #         json.dumps(input_).encode(STRING_ENCODING)
-        #     )
+    return make_request(request.method, endpoint, request.get_json(),
+                        request.args, headers)
 
-        # log.warn('Trying to unpack input:')
-        # log.warn(input_)
 
-        input_unpacked = base64s_to_bytes(input_)
+def make_request(method: str, endpoint: str, json: dict = None,
+                 params: dict = None, headers: dict = None) -> Response:
+    """
+    Make request to the central server
+
+    Parameters
+    ----------
+    method : str
+        HTTP method to be used
+    endpoint : str
+        endpoint of the vantage6-server
+    json : dict, optional
+        JSON body, by default None
+    params : dict, optional
+        HTTP parameters, by default None
+    headers : dict, optional
+        HTTP headers, by default None
+
+    Returns
+    -------
+    Response
+        Response from the vantage6-server
+    """
 
-        encrypted_input = server_io.cryptor.encrypt_bytes_to_str(
-            input_unpacked,
-            public_key
-        )
+    # Obtain http method from request and map it to the same `requests` method
+    method = get_method(request.method)
+
+    # Forward the request to the central server. Retry when an exception is
+    # raised (e.g. timeout or connection error) or when the server gives an
+    # error code greater than 210
+    url = f"{server_url}/{endpoint}"
```

Comment:
check if server_url already ends in '/'?

Reply:
It can't have a trailing '/' as this is comming from the vantage6-client property (`BaseClient.base_path`), many more node actions will fail in this case. 

However you are right, this check is *not* present in the client.. Not sure if we should open a seperate issue for this or that we accept this as a user-input mistake..